runtime/doc/lua.txt

```diff
@@ -3034,6 +3027,363 @@ vim.glob.to_lpeg({pattern})                               *vim.glob.to_lpeg()*
+    the group name as its key. The captured value is only the table.
+
+    Parameters: ~
+      • {patt}  (vim.lpeg.Pattern |' `) @return (` vim.lpeg.Capture`)
+
+vim.lpeg.lpeg                                                     *vim.lpeg()*
+    LPeg is a new pattern-matching library for Lua, based on Parsing Expression
+    Grammars (PEGs).
+
+vim.lpeg.match({pattern}, {subject}, {init})                *vim.lpeg.match()*
+    Matches the given `pattern` against the `subject` string. If the match
+    succeeds, returns the index in the subject of the first character after
+    the match, or the captured values (if the pattern captured any value). An
+    optional numeric argument `init` makes the match start at that position in
+    the subject string. As usual in Lua libraries, a negative value counts
+    from the end. Unlike typical pattern-matching functions, `match` works
+    only in anchored mode; that is, it tries to match the pattern with a
+    prefix of the given subject string (at position `init`), not with an
+    arbitrary substring of the subject. So, if we want to find a pattern
+    anywhere in a string, we must either write a loop in Lua or write a
+    pattern that matches anywhere.
+
+    Example:>lua
+        local pattern = lpeg.R("az") ^ 1 * -1
+        assert(pattern:match("hello") == 6)
+        assert(lpeg.match(pattern, "hello") == 6)
+        assert(pattern:match("1 hello") == nil)
+<
+
+    Parameters: ~
+      • {pattern}  (`vim.lpeg.Pattern`)
+      • {subject}  (`string`)
+      • {init}     (`integer?`)
+
+    Return: ~
+        (`integer|vim.lpeg.Capture?`)
+
+vim.lpeg.P({value})                                             *vim.lpeg.P()*
+    Converts the given value into a proper pattern. This following rules are
+    applied:
+    • If the argument is a pattern, it is returned unmodified.
+    • If the argument is a string, it is translated to a pattern that matches
+      the string literally.
+    • If the argument is a non-negative number `n`, the result is a pattern
+      that matches exactly `n` characters.
+    • If the argument is a negative number `-n`, the result is a pattern that
+      succeeds only if the input string has less than `n` characters left:
+      `lpeg.P(-n)` is equivalent to `-lpeg.P(n)` (see the unary minus
+      operation).
+    • If the argument is a boolean, the result is a pattern that always
+      succeeds or always fails (according to the boolean value), without
+      consuming any input.
+    • If the argument is a table, it is interpreted as a grammar (see
+      Grammars).
+    • If the argument is a function, returns a pattern equivalent to a
+      match-time captureover the empty string.
+
+    Parameters: ~
+      • {value}  (`vim.lpeg.Pattern|string|integer|boolean|table|function`)
+
+    Return: ~
+        (`vim.lpeg.Pattern`)
+
+vim.lpeg.Pattern:match({subject}, {init})                    *Pattern:match()*
```

Comment:
I feel this should be fixed to `*vim.lpeg.Pattern:match()`, it's weird to strip off the fstem prefix. #24363

Reply:
What's sauce for the goose is sauce for the ganter. Either we consistently strip the leading module for objects or we don't -- `vim.lpeg` is not special.

I don't care much either way, but I'd check for tag _links_ to these anchors in our documentation first; that will tell us what effect keeping the module will have.